src/core.js

```diff
@@ -11,6 +11,11 @@ nv.charts = {}; //stores all the ready to use charts
 nv.logs = {}; //stores some statistics and potential error messages
 nv.dom = {}; //DOM manipulation functions
 
+// Node/CommonJS - require D3
+if (typeof(module) !== 'undefined' && typeof(exports) !== 'undefined' && typeof(d3) == 'undefined') {
+    d3 = require('d3');
+}
+
```

Comment:
@m-h-t , the problem is that `nvd3` requires `d3` to be in global space. It's hurt, because global namespacing is a bad pattern.